rpc/client_test.go

```diff
@@ -429,6 +430,41 @@ func TestClientNotificationStorm(t *testing.T) {
 	doTest(23000, true)
 }
 
+// TestClientSetHeader tests whether an http header has been properly set
+// to the given key and value on a client's http request.
+func TestClientSetHeader(t *testing.T) {
```

Comment:
This test is a bit useless because it just checks if the map is updated. If you want to make a real test,
use package httptest to launch a server and verify that the request sent by client contains the custom headers.